<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 6 Parse preudotime data | ClusterGVis Reference Book</title>
  <meta name="description" content="This is a detailed documentation of ClusterGVis package." />
  <meta name="generator" content="bookdown 0.41 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 6 Parse preudotime data | ClusterGVis Reference Book" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a detailed documentation of ClusterGVis package." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 6 Parse preudotime data | ClusterGVis Reference Book" />
  
  <meta name="twitter:description" content="This is a detailed documentation of ClusterGVis package." />
  

<meta name="author" content="Jun Zhang" />


<meta name="date" content="2024-10-25" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="parse-single-cell-data.html"/>
<link rel="next" href="add-custom-annotation.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">ClusterGVis manual documentation</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> About</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#citation"><i class="fa fa-check"></i><b>1.1</b> Citation</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#installation"><i class="fa fa-check"></i><b>1.2</b> Installation</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#bugs"><i class="fa fa-check"></i><b>1.3</b> Bugs</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="basic-usage.html"><a href="basic-usage.html"><i class="fa fa-check"></i><b>2</b> Basic usage</a>
<ul>
<li class="chapter" data-level="2.1" data-path="basic-usage.html"><a href="basic-usage.html#input-data"><i class="fa fa-check"></i><b>2.1</b> Input data</a></li>
<li class="chapter" data-level="2.2" data-path="basic-usage.html"><a href="basic-usage.html#clustering-for-expression-matrix"><i class="fa fa-check"></i><b>2.2</b> Clustering for expression matrix</a></li>
<li class="chapter" data-level="2.3" data-path="basic-usage.html"><a href="basic-usage.html#line-plot"><i class="fa fa-check"></i><b>2.3</b> Line plot</a></li>
<li class="chapter" data-level="2.4" data-path="basic-usage.html"><a href="basic-usage.html#heatmap-plot"><i class="fa fa-check"></i><b>2.4</b> Heatmap plot</a></li>
<li class="chapter" data-level="2.5" data-path="basic-usage.html"><a href="basic-usage.html#sample-group-and-order"><i class="fa fa-check"></i><b>2.5</b> Sample group and order</a></li>
<li class="chapter" data-level="2.6" data-path="basic-usage.html"><a href="basic-usage.html#multiple-sample-annotation"><i class="fa fa-check"></i><b>2.6</b> Multiple sample annotation</a></li>
<li class="chapter" data-level="2.7" data-path="basic-usage.html"><a href="basic-usage.html#and-line-trend-anntation"><i class="fa fa-check"></i><b>2.7</b> And line trend anntation</a></li>
<li class="chapter" data-level="2.8" data-path="basic-usage.html"><a href="basic-usage.html#multiple-group-line-plot"><i class="fa fa-check"></i><b>2.8</b> Multiple group line plot</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="enrichment-anaysis.html"><a href="enrichment-anaysis.html"><i class="fa fa-check"></i><b>3</b> Enrichment anaysis</a>
<ul>
<li class="chapter" data-level="3.1" data-path="enrichment-anaysis.html"><a href="enrichment-anaysis.html#enrichment-for-non-model-organisms"><i class="fa fa-check"></i><b>3.1</b> Enrichment for non-model organisms</a></li>
<li class="chapter" data-level="3.2" data-path="enrichment-anaysis.html"><a href="enrichment-anaysis.html#own-enrichment-data"><i class="fa fa-check"></i><b>3.2</b> Own enrichment data</a></li>
<li class="chapter" data-level="3.3" data-path="enrichment-anaysis.html"><a href="enrichment-anaysis.html#batch-enrichment-analysis"><i class="fa fa-check"></i><b>3.3</b> Batch enrichment analysis</a></li>
<li class="chapter" data-level="3.4" data-path="enrichment-anaysis.html"><a href="enrichment-anaysis.html#add-annotation-for-specific-groups"><i class="fa fa-check"></i><b>3.4</b> Add annotation for specific groups</a></li>
<li class="chapter" data-level="3.5" data-path="enrichment-anaysis.html"><a href="enrichment-anaysis.html#add-barplot-for-enrichment-annotation"><i class="fa fa-check"></i><b>3.5</b> Add barplot for enrichment annotation</a></li>
<li class="chapter" data-level="3.6" data-path="enrichment-anaysis.html"><a href="enrichment-anaysis.html#add-kegg-annotation"><i class="fa fa-check"></i><b>3.6</b> Add KEGG annotation</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="parse-wgcna-output.html"><a href="parse-wgcna-output.html"><i class="fa fa-check"></i><b>4</b> Parse WGCNA output</a></li>
<li class="chapter" data-level="5" data-path="parse-single-cell-data.html"><a href="parse-single-cell-data.html"><i class="fa fa-check"></i><b>5</b> Parse single cell data</a></li>
<li class="chapter" data-level="6" data-path="parse-preudotime-data.html"><a href="parse-preudotime-data.html"><i class="fa fa-check"></i><b>6</b> Parse preudotime data</a>
<ul>
<li class="chapter" data-level="6.1" data-path="parse-preudotime-data.html"><a href="parse-preudotime-data.html#monocle2-implemention"><i class="fa fa-check"></i><b>6.1</b> Monocle2 implemention</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="parse-preudotime-data.html"><a href="parse-preudotime-data.html#plot_pseudotime_heatmap2"><i class="fa fa-check"></i><b>6.1.1</b> plot_pseudotime_heatmap2</a></li>
<li class="chapter" data-level="6.1.2" data-path="parse-preudotime-data.html"><a href="parse-preudotime-data.html#plot_genes_branched_heatmap2"><i class="fa fa-check"></i><b>6.1.2</b> plot_genes_branched_heatmap2</a></li>
<li class="chapter" data-level="6.1.3" data-path="parse-preudotime-data.html"><a href="parse-preudotime-data.html#plot_multiple_branches_heatmap2"><i class="fa fa-check"></i><b>6.1.3</b> plot_multiple_branches_heatmap2</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="parse-preudotime-data.html"><a href="parse-preudotime-data.html#monocle3-implemention"><i class="fa fa-check"></i><b>6.2</b> Monocle3 implemention</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="add-custom-annotation.html"><a href="add-custom-annotation.html"><i class="fa fa-check"></i><b>7</b> Add custom annotation</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">ClusterGVis Reference Book</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="parse-preudotime-data" class="section level1 hasAnchor" number="6">
<h1><span class="header-section-number">Chapter 6</span> Parse preudotime data<a href="parse-preudotime-data.html#parse-preudotime-data" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><strong>Monocle</strong> is a widely used bioinformatics tool designed for analyzing single-cell RNA sequencing (scRNA-seq) data. It allows for trajectory inference, differential expression analysis, and clustering of cells, all of which help in understanding dynamic biological processes such as cell differentiation, development, and lineage relationships. <strong>ClusterGvis</strong> can be implemented with <strong>Monocle</strong> and visualize
the trajectory data.</p>
<div id="monocle2-implemention" class="section level2 hasAnchor" number="6.1">
<h2><span class="header-section-number">6.1</span> Monocle2 implemention<a href="parse-preudotime-data.html#monocle2-implemention" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The functions <strong>plot_pseudotime_heatmap, plot_genes_branched_heatmap, plot_multiple_branches_heatmap</strong> to visualize preudotime data in <strong>monocle2</strong> version are slightly modified into <strong>plot_pseudotime_heatmap2, plot_genes_branched_heatmap2, plot_multiple_branches_heatmap2</strong>. The returned object can be recognized with <strong>visCluster</strong> directly.</p>
<p>The test data is from <strong>monocle::load_lung()</strong>, which you can load directly following the code:</p>
<div class="sourceCode" id="cb58"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb58-1"><a href="parse-preudotime-data.html#cb58-1" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;HSMM&quot;</span>)</span>
<span id="cb58-2"><a href="parse-preudotime-data.html#cb58-2" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;BEAM_res&quot;</span>)</span>
<span id="cb58-3"><a href="parse-preudotime-data.html#cb58-3" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;sig_gene_names&quot;</span>)</span></code></pre></div>
<div id="plot_pseudotime_heatmap2" class="section level3 hasAnchor" number="6.1.1">
<h3><span class="header-section-number">6.1.1</span> plot_pseudotime_heatmap2<a href="parse-preudotime-data.html#plot_pseudotime_heatmap2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Return heatmap only using <strong>return_heatmap=T</strong>:</p>
<div class="sourceCode" id="cb59"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb59-1"><a href="parse-preudotime-data.html#cb59-1" tabindex="-1"></a><span class="co"># return plot</span></span>
<span id="cb59-2"><a href="parse-preudotime-data.html#cb59-2" tabindex="-1"></a><span class="fu">plot_pseudotime_heatmap2</span>(HSMM[sig_gene_names[<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>],],</span>
<span id="cb59-3"><a href="parse-preudotime-data.html#cb59-3" tabindex="-1"></a>                         <span class="at">num_clusters =</span> <span class="dv">4</span>,</span>
<span id="cb59-4"><a href="parse-preudotime-data.html#cb59-4" tabindex="-1"></a>                         <span class="at">cores =</span> <span class="dv">1</span>,</span>
<span id="cb59-5"><a href="parse-preudotime-data.html#cb59-5" tabindex="-1"></a>                         <span class="at">show_rownames =</span> T,</span>
<span id="cb59-6"><a href="parse-preudotime-data.html#cb59-6" tabindex="-1"></a>                         <span class="at">return_heatmap =</span> T)</span></code></pre></div>
<p align="center">
<img src="local_images/34.png" width="80%">
</p>
<p>Return list object containing clustered data:</p>
<div class="sourceCode" id="cb60"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb60-1"><a href="parse-preudotime-data.html#cb60-1" tabindex="-1"></a><span class="co"># test code</span></span>
<span id="cb60-2"><a href="parse-preudotime-data.html#cb60-2" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">plot_pseudotime_heatmap2</span>(HSMM[sig_gene_names[<span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>],],</span>
<span id="cb60-3"><a href="parse-preudotime-data.html#cb60-3" tabindex="-1"></a>                               <span class="at">num_clusters =</span> <span class="dv">4</span>,</span>
<span id="cb60-4"><a href="parse-preudotime-data.html#cb60-4" tabindex="-1"></a>                               <span class="at">cores =</span> <span class="dv">1</span>)</span>
<span id="cb60-5"><a href="parse-preudotime-data.html#cb60-5" tabindex="-1"></a></span>
<span id="cb60-6"><a href="parse-preudotime-data.html#cb60-6" tabindex="-1"></a><span class="co"># check data</span></span>
<span id="cb60-7"><a href="parse-preudotime-data.html#cb60-7" tabindex="-1"></a><span class="fu">str</span>(df)</span>
<span id="cb60-8"><a href="parse-preudotime-data.html#cb60-8" tabindex="-1"></a><span class="co"># List of 6</span></span>
<span id="cb60-9"><a href="parse-preudotime-data.html#cb60-9" tabindex="-1"></a><span class="co"># $ wide.res  :&#39;data.frame&#39;:    100 obs. of  102 variables:</span></span>
<span id="cb60-10"><a href="parse-preudotime-data.html#cb60-10" tabindex="-1"></a><span class="co">#   ..$ gene   : chr [1:100] &quot;TSPAN6&quot; &quot;C1orf112&quot; &quot;CYP51A1&quot; &quot;AK2&quot; ...</span></span>
<span id="cb60-11"><a href="parse-preudotime-data.html#cb60-11" tabindex="-1"></a><span class="co"># ..$ 1      : num [1:100] -1.573 2.783 -1.646 -0.566 2.264 ...</span></span>
<span id="cb60-12"><a href="parse-preudotime-data.html#cb60-12" tabindex="-1"></a><span class="co"># ..$ 2      : num [1:100] -1.55 2.65 -1.62 -0.57 2.19 ...</span></span>
<span id="cb60-13"><a href="parse-preudotime-data.html#cb60-13" tabindex="-1"></a><span class="co"># ..$ 3      : num [1:100] -1.524 2.517 -1.587 -0.575 2.118 ...</span></span>
<span id="cb60-14"><a href="parse-preudotime-data.html#cb60-14" tabindex="-1"></a><span class="co"># ...</span></span>
<span id="cb60-15"><a href="parse-preudotime-data.html#cb60-15" tabindex="-1"></a><span class="co"># ..$ 96     : num [1:100] 0.48 -1.421 0.363 2.647 -1.753 ...</span></span>
<span id="cb60-16"><a href="parse-preudotime-data.html#cb60-16" tabindex="-1"></a><span class="co"># ..$ 97     : num [1:100] 0.34 -1.46 0.21 2.86 -1.94 ...</span></span>
<span id="cb60-17"><a href="parse-preudotime-data.html#cb60-17" tabindex="-1"></a><span class="co"># ..$ 98     : num [1:100] 0.1988 -1.4958 0.0542 3 -2.1274 ...</span></span>
<span id="cb60-18"><a href="parse-preudotime-data.html#cb60-18" tabindex="-1"></a><span class="co"># .. [list output truncated]</span></span>
<span id="cb60-19"><a href="parse-preudotime-data.html#cb60-19" tabindex="-1"></a><span class="co"># $ long.res  :&#39;data.frame&#39;:    10000 obs. of  5 variables:</span></span>
<span id="cb60-20"><a href="parse-preudotime-data.html#cb60-20" tabindex="-1"></a><span class="co">#   ..$ cluster     : Factor w/ 4 levels &quot;1&quot;,&quot;2&quot;,&quot;3&quot;,&quot;4&quot;: 1 1 1 1 1 1 1 1 1 1 ...</span></span>
<span id="cb60-21"><a href="parse-preudotime-data.html#cb60-21" tabindex="-1"></a><span class="co"># ..$ gene        : chr [1:10000] &quot;TSPAN6&quot; &quot;CYP51A1&quot; &quot;AASS&quot; &quot;MGST1&quot; ...</span></span>
<span id="cb60-22"><a href="parse-preudotime-data.html#cb60-22" tabindex="-1"></a><span class="co"># ..$ cell_type   : num [1:10000] 1 1 1 1 1 1 1 1 1 1 ...</span></span>
<span id="cb60-23"><a href="parse-preudotime-data.html#cb60-23" tabindex="-1"></a><span class="co"># ..$ norm_value  : num [1:10000] -1.573 -1.646 -1.86 -0.457 -2.024 ...</span></span>
<span id="cb60-24"><a href="parse-preudotime-data.html#cb60-24" tabindex="-1"></a><span class="co"># ..$ cluster_name: Factor w/ 4 levels &quot;cluster 1 (14)&quot;,..: 1 1 1 1 1 1 1 1 1 1 ...</span></span>
<span id="cb60-25"><a href="parse-preudotime-data.html#cb60-25" tabindex="-1"></a><span class="co"># $ type      : chr &quot;monocle&quot;</span></span>
<span id="cb60-26"><a href="parse-preudotime-data.html#cb60-26" tabindex="-1"></a><span class="co"># $ geneMode  : chr &quot;all&quot;</span></span>
<span id="cb60-27"><a href="parse-preudotime-data.html#cb60-27" tabindex="-1"></a><span class="co"># $ geneType  : chr &quot;non-branched&quot;</span></span>
<span id="cb60-28"><a href="parse-preudotime-data.html#cb60-28" tabindex="-1"></a><span class="co"># $ pseudotime: num [1:100] 0 0.236 0.471 0.707 0.942 ...</span></span></code></pre></div>
<p>Line plot:</p>
<div class="sourceCode" id="cb61"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb61-1"><a href="parse-preudotime-data.html#cb61-1" tabindex="-1"></a><span class="co"># plot</span></span>
<span id="cb61-2"><a href="parse-preudotime-data.html#cb61-2" tabindex="-1"></a><span class="fu">visCluster</span>(<span class="at">object =</span> df,<span class="at">plot.type =</span> <span class="st">&quot;line&quot;</span>)</span></code></pre></div>
<p align="center">
<img src="local_images/35.png" width="80%">
</p>
<p>Heatmap:</p>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="parse-preudotime-data.html#cb62-1" tabindex="-1"></a><span class="fu">visCluster</span>(<span class="at">object =</span> df,<span class="at">plot.type =</span> <span class="st">&quot;heatmap&quot;</span>)</span></code></pre></div>
<p align="center">
<img src="local_images/36.png" width="80%">
</p>
<p>Line annotation and mark gene names:</p>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="parse-preudotime-data.html#cb63-1" tabindex="-1"></a>gene <span class="ot">=</span> <span class="fu">sample</span>(df<span class="sc">$</span>wide.res<span class="sc">$</span>gene,<span class="dv">20</span>,<span class="at">replace =</span> F)</span>
<span id="cb63-2"><a href="parse-preudotime-data.html#cb63-2" tabindex="-1"></a></span>
<span id="cb63-3"><a href="parse-preudotime-data.html#cb63-3" tabindex="-1"></a><span class="fu">pdf</span>(<span class="at">file =</span> <span class="st">&quot;one-branch.pdf&quot;</span>,<span class="at">height =</span> <span class="dv">6</span>,<span class="at">width =</span> <span class="dv">7</span>)</span>
<span id="cb63-4"><a href="parse-preudotime-data.html#cb63-4" tabindex="-1"></a><span class="fu">visCluster</span>(<span class="at">object =</span> df,</span>
<span id="cb63-5"><a href="parse-preudotime-data.html#cb63-5" tabindex="-1"></a>           <span class="at">plot.type =</span> <span class="st">&quot;both&quot;</span>,</span>
<span id="cb63-6"><a href="parse-preudotime-data.html#cb63-6" tabindex="-1"></a>           <span class="at">markGenes =</span> gene)</span>
<span id="cb63-7"><a href="parse-preudotime-data.html#cb63-7" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
<p align="center">
<img src="local_images/37.png" width="80%">
</p>
</div>
<div id="plot_genes_branched_heatmap2" class="section level3 hasAnchor" number="6.1.2">
<h3><span class="header-section-number">6.1.2</span> plot_genes_branched_heatmap2<a href="parse-preudotime-data.html#plot_genes_branched_heatmap2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Return plot only:</p>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="parse-preudotime-data.html#cb64-1" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;HSMM&quot;</span>)</span>
<span id="cb64-2"><a href="parse-preudotime-data.html#cb64-2" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;BEAM_res&quot;</span>)</span>
<span id="cb64-3"><a href="parse-preudotime-data.html#cb64-3" tabindex="-1"></a></span>
<span id="cb64-4"><a href="parse-preudotime-data.html#cb64-4" tabindex="-1"></a><span class="co"># return plot</span></span>
<span id="cb64-5"><a href="parse-preudotime-data.html#cb64-5" tabindex="-1"></a><span class="fu">plot_genes_branched_heatmap2</span>(HSMM[<span class="fu">row.names</span>(<span class="fu">subset</span>(BEAM_res,qval <span class="sc">&lt;</span> <span class="fl">1e-4</span>)),],</span>
<span id="cb64-6"><a href="parse-preudotime-data.html#cb64-6" tabindex="-1"></a>                             <span class="at">branch_point =</span> <span class="dv">1</span>,</span>
<span id="cb64-7"><a href="parse-preudotime-data.html#cb64-7" tabindex="-1"></a>                             <span class="at">num_clusters =</span> <span class="dv">4</span>,</span>
<span id="cb64-8"><a href="parse-preudotime-data.html#cb64-8" tabindex="-1"></a>                             <span class="at">cores =</span> <span class="dv">1</span>,</span>
<span id="cb64-9"><a href="parse-preudotime-data.html#cb64-9" tabindex="-1"></a>                             <span class="at">use_gene_short_name =</span> T,</span>
<span id="cb64-10"><a href="parse-preudotime-data.html#cb64-10" tabindex="-1"></a>                             <span class="at">show_rownames =</span> T,</span>
<span id="cb64-11"><a href="parse-preudotime-data.html#cb64-11" tabindex="-1"></a>                             <span class="at">return_heatmap =</span> T)</span></code></pre></div>
<p align="center">
<img src="local_images/38.png" width="80%">
</p>
<p>Return data:</p>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="parse-preudotime-data.html#cb65-1" tabindex="-1"></a><span class="co"># test code</span></span>
<span id="cb65-2"><a href="parse-preudotime-data.html#cb65-2" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">plot_genes_branched_heatmap2</span>(HSMM[<span class="fu">row.names</span>(<span class="fu">subset</span>(BEAM_res,qval <span class="sc">&lt;</span> <span class="fl">1e-4</span>)),],</span>
<span id="cb65-3"><a href="parse-preudotime-data.html#cb65-3" tabindex="-1"></a>                                   <span class="at">branch_point =</span> <span class="dv">1</span>,</span>
<span id="cb65-4"><a href="parse-preudotime-data.html#cb65-4" tabindex="-1"></a>                                   <span class="at">num_clusters =</span> <span class="dv">4</span>,</span>
<span id="cb65-5"><a href="parse-preudotime-data.html#cb65-5" tabindex="-1"></a>                                   <span class="at">cores =</span> <span class="dv">1</span>,</span>
<span id="cb65-6"><a href="parse-preudotime-data.html#cb65-6" tabindex="-1"></a>                                   <span class="at">use_gene_short_name =</span> T,</span>
<span id="cb65-7"><a href="parse-preudotime-data.html#cb65-7" tabindex="-1"></a>                                   <span class="at">show_rownames =</span> T)</span>
<span id="cb65-8"><a href="parse-preudotime-data.html#cb65-8" tabindex="-1"></a></span>
<span id="cb65-9"><a href="parse-preudotime-data.html#cb65-9" tabindex="-1"></a><span class="co"># check data</span></span>
<span id="cb65-10"><a href="parse-preudotime-data.html#cb65-10" tabindex="-1"></a><span class="fu">str</span>(df)</span>
<span id="cb65-11"><a href="parse-preudotime-data.html#cb65-11" tabindex="-1"></a><span class="co"># List of 6</span></span>
<span id="cb65-12"><a href="parse-preudotime-data.html#cb65-12" tabindex="-1"></a><span class="co"># $ wide.res  :&#39;data.frame&#39;: 36 obs. of  202 variables:</span></span>
<span id="cb65-13"><a href="parse-preudotime-data.html#cb65-13" tabindex="-1"></a><span class="co">#   ..$ gene   : chr [1:36] &quot;BIRC5&quot; &quot;CDKN3&quot; &quot;ACTA2&quot; &quot;EFEMP1&quot; ...</span></span>
<span id="cb65-14"><a href="parse-preudotime-data.html#cb65-14" tabindex="-1"></a><span class="co"># ..$ 1      : num [1:36] 1.544 2.571 1.329 -1.595 0.609 ...</span></span>
<span id="cb65-15"><a href="parse-preudotime-data.html#cb65-15" tabindex="-1"></a><span class="co"># ..$ 2      : num [1:36] 1.317 2.021 1.17 -1.45 0.498 ...</span></span>
<span id="cb65-16"><a href="parse-preudotime-data.html#cb65-16" tabindex="-1"></a><span class="co"># ..$ 3      : num [1:36] 1.101 1.505 1.015 -1.295 0.389 ...</span></span>
<span id="cb65-17"><a href="parse-preudotime-data.html#cb65-17" tabindex="-1"></a><span class="co"># .. [list output truncated]</span></span>
<span id="cb65-18"><a href="parse-preudotime-data.html#cb65-18" tabindex="-1"></a><span class="co"># $ long.res  :&#39;data.frame&#39;: 7200 obs. of  5 variables:</span></span>
<span id="cb65-19"><a href="parse-preudotime-data.html#cb65-19" tabindex="-1"></a><span class="co">#   ..$ cluster     : Factor w/ 4 levels &quot;1&quot;,&quot;2&quot;,&quot;3&quot;,&quot;4&quot;: 1 1 1 1 1 1 1 1 1 1 ...</span></span>
<span id="cb65-20"><a href="parse-preudotime-data.html#cb65-20" tabindex="-1"></a><span class="co"># ..$ gene        : chr [1:7200] &quot;BIRC5&quot; &quot;CDKN3&quot; &quot;STMN1&quot; &quot;CENPF&quot; ...</span></span>
<span id="cb65-21"><a href="parse-preudotime-data.html#cb65-21" tabindex="-1"></a><span class="co"># ..$ cell_type   : num [1:7200] 1 1 1 1 1 1 1 1 1 1 ...</span></span>
<span id="cb65-22"><a href="parse-preudotime-data.html#cb65-22" tabindex="-1"></a><span class="co"># ..$ norm_value  : num [1:7200] 1.544 2.571 0.609 1.807 2.177 ...</span></span>
<span id="cb65-23"><a href="parse-preudotime-data.html#cb65-23" tabindex="-1"></a><span class="co"># ..$ cluster_name: Factor w/ 4 levels &quot;cluster 1 (27)&quot;,..: 1 1 1 1 1 1 1 1 1 1 ...</span></span>
<span id="cb65-24"><a href="parse-preudotime-data.html#cb65-24" tabindex="-1"></a><span class="co"># $ type      : chr &quot;monocle&quot;</span></span>
<span id="cb65-25"><a href="parse-preudotime-data.html#cb65-25" tabindex="-1"></a><span class="co"># $ geneMode  : chr &quot;all&quot;</span></span>
<span id="cb65-26"><a href="parse-preudotime-data.html#cb65-26" tabindex="-1"></a><span class="co"># $ geneType  : chr &quot;branched&quot;</span></span>
<span id="cb65-27"><a href="parse-preudotime-data.html#cb65-27" tabindex="-1"></a><span class="co"># $ pseudotime: chr [1:200] &quot;Cell fate 1&quot; &quot;Cell fate 1&quot; &quot;Cell fate 1&quot; &quot;Cell fate 1&quot; ...</span></span></code></pre></div>
<p>Heatmap:</p>
<div class="sourceCode" id="cb66"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb66-1"><a href="parse-preudotime-data.html#cb66-1" tabindex="-1"></a><span class="fu">visCluster</span>(<span class="at">object =</span> df,<span class="at">plot.type =</span> <span class="st">&quot;heatmap&quot;</span>)</span></code></pre></div>
<p align="center">
<img src="local_images/39.png" width="80%">
</p>
<p>Line annotation:</p>
<div class="sourceCode" id="cb67"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb67-1"><a href="parse-preudotime-data.html#cb67-1" tabindex="-1"></a><span class="fu">pdf</span>(<span class="at">file =</span> <span class="st">&quot;two-branch.pdf&quot;</span>,<span class="at">height =</span> <span class="dv">6</span>,<span class="at">width =</span> <span class="dv">7</span>)</span>
<span id="cb67-2"><a href="parse-preudotime-data.html#cb67-2" tabindex="-1"></a><span class="fu">visCluster</span>(<span class="at">object =</span> df,</span>
<span id="cb67-3"><a href="parse-preudotime-data.html#cb67-3" tabindex="-1"></a>           <span class="at">plot.type =</span> <span class="st">&quot;both&quot;</span>)</span>
<span id="cb67-4"><a href="parse-preudotime-data.html#cb67-4" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
<p align="center">
<img src="local_images/40.png" width="80%">
</p>
</div>
<div id="plot_multiple_branches_heatmap2" class="section level3 hasAnchor" number="6.1.3">
<h3><span class="header-section-number">6.1.3</span> plot_multiple_branches_heatmap2<a href="parse-preudotime-data.html#plot_multiple_branches_heatmap2" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Return plot only:</p>
<div class="sourceCode" id="cb68"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb68-1"><a href="parse-preudotime-data.html#cb68-1" tabindex="-1"></a><span class="co"># return plot</span></span>
<span id="cb68-2"><a href="parse-preudotime-data.html#cb68-2" tabindex="-1"></a><span class="fu">plot_multiple_branches_heatmap2</span>(HSMM[<span class="fu">row.names</span>(<span class="fu">subset</span>(BEAM_res,qval <span class="sc">&lt;</span> <span class="fl">1e-4</span>)),],</span>
<span id="cb68-3"><a href="parse-preudotime-data.html#cb68-3" tabindex="-1"></a>                                <span class="at">branches =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">5</span>),</span>
<span id="cb68-4"><a href="parse-preudotime-data.html#cb68-4" tabindex="-1"></a>                                <span class="at">num_clusters =</span> <span class="dv">4</span>,</span>
<span id="cb68-5"><a href="parse-preudotime-data.html#cb68-5" tabindex="-1"></a>                                <span class="at">cores =</span> <span class="dv">1</span>,</span>
<span id="cb68-6"><a href="parse-preudotime-data.html#cb68-6" tabindex="-1"></a>                                <span class="at">use_gene_short_name =</span> T,</span>
<span id="cb68-7"><a href="parse-preudotime-data.html#cb68-7" tabindex="-1"></a>                                <span class="at">show_rownames =</span> T,</span>
<span id="cb68-8"><a href="parse-preudotime-data.html#cb68-8" tabindex="-1"></a>                                <span class="at">return_heatmap =</span> T)</span></code></pre></div>
<p align="center">
<img src="local_images/41.png" width="80%">
</p>
<p>Return data:</p>
<div class="sourceCode" id="cb69"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb69-1"><a href="parse-preudotime-data.html#cb69-1" tabindex="-1"></a><span class="co"># test code</span></span>
<span id="cb69-2"><a href="parse-preudotime-data.html#cb69-2" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">plot_multiple_branches_heatmap2</span>(HSMM[<span class="fu">row.names</span>(<span class="fu">subset</span>(BEAM_res,qval <span class="sc">&lt;</span> <span class="fl">1e-4</span>)),],</span>
<span id="cb69-3"><a href="parse-preudotime-data.html#cb69-3" tabindex="-1"></a>                                      <span class="at">branches =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">5</span>),</span>
<span id="cb69-4"><a href="parse-preudotime-data.html#cb69-4" tabindex="-1"></a>                                      <span class="at">num_clusters =</span> <span class="dv">4</span>,</span>
<span id="cb69-5"><a href="parse-preudotime-data.html#cb69-5" tabindex="-1"></a>                                      <span class="at">cores =</span> <span class="dv">1</span>,</span>
<span id="cb69-6"><a href="parse-preudotime-data.html#cb69-6" tabindex="-1"></a>                                      <span class="at">use_gene_short_name =</span> T,</span>
<span id="cb69-7"><a href="parse-preudotime-data.html#cb69-7" tabindex="-1"></a>                                      <span class="at">show_rownames =</span> T)</span>
<span id="cb69-8"><a href="parse-preudotime-data.html#cb69-8" tabindex="-1"></a></span>
<span id="cb69-9"><a href="parse-preudotime-data.html#cb69-9" tabindex="-1"></a><span class="co"># check data</span></span>
<span id="cb69-10"><a href="parse-preudotime-data.html#cb69-10" tabindex="-1"></a><span class="fu">str</span>(df)</span>
<span id="cb69-11"><a href="parse-preudotime-data.html#cb69-11" tabindex="-1"></a><span class="co"># List of 6</span></span>
<span id="cb69-12"><a href="parse-preudotime-data.html#cb69-12" tabindex="-1"></a><span class="co"># $ wide.res  :&#39;data.frame&#39;: 36 obs. of  402 variables:</span></span>
<span id="cb69-13"><a href="parse-preudotime-data.html#cb69-13" tabindex="-1"></a><span class="co">#   ..$ gene   : chr [1:36] &quot;BIRC5&quot; &quot;CDKN3&quot; &quot;ACTA2&quot; &quot;EFEMP1&quot; ...</span></span>
<span id="cb69-14"><a href="parse-preudotime-data.html#cb69-14" tabindex="-1"></a><span class="co"># ..$ 1      : num [1:36] 2.124 2.219 -0.908 -1.749 1.244 ...</span></span>
<span id="cb69-15"><a href="parse-preudotime-data.html#cb69-15" tabindex="-1"></a><span class="co"># ..$ 2      : num [1:36] 2.061 2.156 -0.937 -1.699 1.24 ...</span></span>
<span id="cb69-16"><a href="parse-preudotime-data.html#cb69-16" tabindex="-1"></a><span class="co"># ..$ 3      : num [1:36] 1.999 2.094 -0.965 -1.649 1.236 ...</span></span>
<span id="cb69-17"><a href="parse-preudotime-data.html#cb69-17" tabindex="-1"></a><span class="co"># $ long.res  :&#39;data.frame&#39;: 14400 obs. of  5 variables:</span></span>
<span id="cb69-18"><a href="parse-preudotime-data.html#cb69-18" tabindex="-1"></a><span class="co">#   ..$ cluster     : Factor w/ 4 levels &quot;1&quot;,&quot;2&quot;,&quot;3&quot;,&quot;4&quot;: 1 1 1 1 1 1 1 1 1 1 ...</span></span>
<span id="cb69-19"><a href="parse-preudotime-data.html#cb69-19" tabindex="-1"></a><span class="co"># ..$ gene        : chr [1:14400] &quot;BIRC5&quot; &quot;CDKN3&quot; &quot;STMN1&quot; &quot;CENPF&quot; ...</span></span>
<span id="cb69-20"><a href="parse-preudotime-data.html#cb69-20" tabindex="-1"></a><span class="co"># ..$ cell_type   : num [1:14400] 1 1 1 1 1 1 1 1 1 1 ...</span></span>
<span id="cb69-21"><a href="parse-preudotime-data.html#cb69-21" tabindex="-1"></a><span class="co"># ..$ norm_value  : num [1:14400] 2.12 2.22 1.24 2.01 2.71 ...</span></span>
<span id="cb69-22"><a href="parse-preudotime-data.html#cb69-22" tabindex="-1"></a><span class="co"># ..$ cluster_name: Factor w/ 4 levels &quot;cluster 1 (18)&quot;,..: 1 1 1 1 1 1 1 1 1 1 ...</span></span>
<span id="cb69-23"><a href="parse-preudotime-data.html#cb69-23" tabindex="-1"></a><span class="co"># $ type      : chr &quot;monocle&quot;</span></span>
<span id="cb69-24"><a href="parse-preudotime-data.html#cb69-24" tabindex="-1"></a><span class="co"># $ geneMode  : chr &quot;all&quot;</span></span>
<span id="cb69-25"><a href="parse-preudotime-data.html#cb69-25" tabindex="-1"></a><span class="co"># $ geneType  : chr &quot;branched&quot;</span></span>
<span id="cb69-26"><a href="parse-preudotime-data.html#cb69-26" tabindex="-1"></a><span class="co"># $ pseudotime: Factor w/ 4 levels &quot;1&quot;,&quot;3&quot;,&quot;4&quot;,&quot;5&quot;: 1 1 1 1 1 1 1 1 1 1 ...</span></span></code></pre></div>
<p>Heatmap:</p>
<div class="sourceCode" id="cb70"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb70-1"><a href="parse-preudotime-data.html#cb70-1" tabindex="-1"></a><span class="fu">visCluster</span>(<span class="at">object =</span> df,<span class="at">plot.type =</span> <span class="st">&quot;heatmap&quot;</span>)</span></code></pre></div>
<p align="center">
<img src="local_images/42.png" width="80%">
</p>
<div class="sourceCode" id="cb71"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb71-1"><a href="parse-preudotime-data.html#cb71-1" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb71-2"><a href="parse-preudotime-data.html#cb71-2" tabindex="-1"></a></span>
<span id="cb71-3"><a href="parse-preudotime-data.html#cb71-3" tabindex="-1"></a><span class="fu">pdf</span>(<span class="at">file =</span> <span class="st">&quot;multiple-branch.pdf&quot;</span>,<span class="at">height =</span> <span class="dv">6</span>,<span class="at">width =</span> <span class="dv">8</span>)</span>
<span id="cb71-4"><a href="parse-preudotime-data.html#cb71-4" tabindex="-1"></a><span class="fu">visCluster</span>(<span class="at">object =</span> df,</span>
<span id="cb71-5"><a href="parse-preudotime-data.html#cb71-5" tabindex="-1"></a>           <span class="at">plot.type =</span> <span class="st">&quot;both&quot;</span>,</span>
<span id="cb71-6"><a href="parse-preudotime-data.html#cb71-6" tabindex="-1"></a>           <span class="at">ht.col.list =</span> <span class="fu">list</span>(<span class="at">col_range =</span> <span class="fu">seq</span>(<span class="sc">-</span><span class="dv">2</span>,<span class="dv">2</span>,<span class="at">length.out =</span> <span class="dv">100</span>),</span>
<span id="cb71-7"><a href="parse-preudotime-data.html#cb71-7" tabindex="-1"></a>                              <span class="at">col_color =</span> <span class="fu">colorRampPalette</span>(<span class="fu">brewer.pal</span>(<span class="dv">9</span>,<span class="st">&quot;PRGn&quot;</span>))(<span class="dv">100</span>)),</span>
<span id="cb71-8"><a href="parse-preudotime-data.html#cb71-8" tabindex="-1"></a>           <span class="at">pseudotime_col =</span> ggsci<span class="sc">::</span><span class="fu">pal_d3</span>()(<span class="dv">4</span>))</span>
<span id="cb71-9"><a href="parse-preudotime-data.html#cb71-9" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
<p align="center">
<img src="local_images/43.png" width="80%">
</p>
<p>Add enrichment annotation for each cluster:</p>
<div class="sourceCode" id="cb72"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb72-1"><a href="parse-preudotime-data.html#cb72-1" tabindex="-1"></a><span class="co"># enrich</span></span>
<span id="cb72-2"><a href="parse-preudotime-data.html#cb72-2" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">plot_multiple_branches_heatmap2</span>(HSMM[<span class="fu">row.names</span>(<span class="fu">subset</span>(BEAM_res,qval <span class="sc">&lt;</span> <span class="fl">0.1</span>)),],</span>
<span id="cb72-3"><a href="parse-preudotime-data.html#cb72-3" tabindex="-1"></a>                                      <span class="at">branches =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">3</span>,<span class="dv">4</span>,<span class="dv">5</span>),</span>
<span id="cb72-4"><a href="parse-preudotime-data.html#cb72-4" tabindex="-1"></a>                                      <span class="at">num_clusters =</span> <span class="dv">4</span>,</span>
<span id="cb72-5"><a href="parse-preudotime-data.html#cb72-5" tabindex="-1"></a>                                      <span class="at">cores =</span> <span class="dv">1</span>,</span>
<span id="cb72-6"><a href="parse-preudotime-data.html#cb72-6" tabindex="-1"></a>                                      <span class="at">use_gene_short_name =</span> T,</span>
<span id="cb72-7"><a href="parse-preudotime-data.html#cb72-7" tabindex="-1"></a>                                      <span class="at">show_rownames =</span> T)</span>
<span id="cb72-8"><a href="parse-preudotime-data.html#cb72-8" tabindex="-1"></a></span>
<span id="cb72-9"><a href="parse-preudotime-data.html#cb72-9" tabindex="-1"></a><span class="co"># enrich for clusters</span></span>
<span id="cb72-10"><a href="parse-preudotime-data.html#cb72-10" tabindex="-1"></a><span class="fu">library</span>(org.Hs.eg.db)</span>
<span id="cb72-11"><a href="parse-preudotime-data.html#cb72-11" tabindex="-1"></a>enrich <span class="ot">&lt;-</span> <span class="fu">enrichCluster</span>(<span class="at">object =</span> df,</span>
<span id="cb72-12"><a href="parse-preudotime-data.html#cb72-12" tabindex="-1"></a>                        <span class="at">OrgDb =</span> org.Hs.eg.db,</span>
<span id="cb72-13"><a href="parse-preudotime-data.html#cb72-13" tabindex="-1"></a>                        <span class="at">type =</span> <span class="st">&quot;BP&quot;</span>,</span>
<span id="cb72-14"><a href="parse-preudotime-data.html#cb72-14" tabindex="-1"></a>                        <span class="at">organism =</span> <span class="st">&quot;hsa&quot;</span>,</span>
<span id="cb72-15"><a href="parse-preudotime-data.html#cb72-15" tabindex="-1"></a>                        <span class="at">pvalueCutoff =</span> <span class="fl">0.5</span>,</span>
<span id="cb72-16"><a href="parse-preudotime-data.html#cb72-16" tabindex="-1"></a>                        <span class="at">topn =</span> <span class="dv">5</span>,</span>
<span id="cb72-17"><a href="parse-preudotime-data.html#cb72-17" tabindex="-1"></a>                        <span class="at">seed =</span> <span class="dv">5201314</span>)</span>
<span id="cb72-18"><a href="parse-preudotime-data.html#cb72-18" tabindex="-1"></a></span>
<span id="cb72-19"><a href="parse-preudotime-data.html#cb72-19" tabindex="-1"></a><span class="co"># check</span></span>
<span id="cb72-20"><a href="parse-preudotime-data.html#cb72-20" tabindex="-1"></a><span class="fu">head</span>(enrich[<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>,])</span>
<span id="cb72-21"><a href="parse-preudotime-data.html#cb72-21" tabindex="-1"></a></span>
<span id="cb72-22"><a href="parse-preudotime-data.html#cb72-22" tabindex="-1"></a><span class="co">#            group                      Description      pvalue ratio</span></span>
<span id="cb72-23"><a href="parse-preudotime-data.html#cb72-23" tabindex="-1"></a><span class="co"># GO:1904238    C1    pericyte cell differentiation 0.002325826    25</span></span>
<span id="cb72-24"><a href="parse-preudotime-data.html#cb72-24" tabindex="-1"></a><span class="co"># GO:0034982    C1 mitochondrial protein processing 0.002748268    25</span></span>
<span id="cb72-25"><a href="parse-preudotime-data.html#cb72-25" tabindex="-1"></a><span class="co"># GO:0072537    C1            fibroblast activation 0.002748268    25</span></span>
<span id="cb72-26"><a href="parse-preudotime-data.html#cb72-26" tabindex="-1"></a></span>
<span id="cb72-27"><a href="parse-preudotime-data.html#cb72-27" tabindex="-1"></a>markGenes <span class="ot">=</span> <span class="fu">sample</span>(<span class="fu">unique</span>(df<span class="sc">$</span>wide.res<span class="sc">$</span>gene),<span class="dv">25</span>,<span class="at">replace =</span> F)</span>
<span id="cb72-28"><a href="parse-preudotime-data.html#cb72-28" tabindex="-1"></a></span>
<span id="cb72-29"><a href="parse-preudotime-data.html#cb72-29" tabindex="-1"></a><span class="co"># PLOT</span></span>
<span id="cb72-30"><a href="parse-preudotime-data.html#cb72-30" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">&#39;branch-enrich.pdf&#39;</span>,<span class="at">height =</span> <span class="dv">9</span>,<span class="at">width =</span> <span class="dv">16</span>,<span class="at">onefile =</span> F)</span>
<span id="cb72-31"><a href="parse-preudotime-data.html#cb72-31" tabindex="-1"></a><span class="fu">visCluster</span>(<span class="at">object =</span> df,</span>
<span id="cb72-32"><a href="parse-preudotime-data.html#cb72-32" tabindex="-1"></a>           <span class="at">plot.type =</span> <span class="st">&quot;both&quot;</span>,</span>
<span id="cb72-33"><a href="parse-preudotime-data.html#cb72-33" tabindex="-1"></a>           <span class="at">column_names_rot =</span> <span class="dv">45</span>,</span>
<span id="cb72-34"><a href="parse-preudotime-data.html#cb72-34" tabindex="-1"></a>           <span class="at">show_row_dend =</span> F,</span>
<span id="cb72-35"><a href="parse-preudotime-data.html#cb72-35" tabindex="-1"></a>           <span class="at">markGenes =</span> markGenes,</span>
<span id="cb72-36"><a href="parse-preudotime-data.html#cb72-36" tabindex="-1"></a>           <span class="at">markGenes.side =</span> <span class="st">&quot;left&quot;</span>,</span>
<span id="cb72-37"><a href="parse-preudotime-data.html#cb72-37" tabindex="-1"></a>           <span class="at">annoTerm.data =</span> enrich,</span>
<span id="cb72-38"><a href="parse-preudotime-data.html#cb72-38" tabindex="-1"></a>           <span class="at">go.col =</span> <span class="fu">rep</span>(jjAnno<span class="sc">::</span><span class="fu">useMyCol</span>(<span class="st">&quot;calm&quot;</span>,<span class="at">n =</span> <span class="dv">4</span>),<span class="at">each =</span> <span class="dv">5</span>),</span>
<span id="cb72-39"><a href="parse-preudotime-data.html#cb72-39" tabindex="-1"></a>           <span class="at">add.bar =</span> T,</span>
<span id="cb72-40"><a href="parse-preudotime-data.html#cb72-40" tabindex="-1"></a>           <span class="at">line.side =</span> <span class="st">&quot;left&quot;</span>)</span>
<span id="cb72-41"><a href="parse-preudotime-data.html#cb72-41" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
<p align="center">
<img src="local_images/44.png" width="80%">
</p>
</div>
</div>
<div id="monocle3-implemention" class="section level2 hasAnchor" number="6.2">
<h2><span class="header-section-number">6.2</span> Monocle3 implemention<a href="parse-preudotime-data.html#monocle3-implemention" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>There is a big update for <strong>monocle3</strong> and much of functions have been removed from <strong>monocle3</strong>. We supply <strong>pre_pseudotime_matrix</strong> function to extract differentiated genes and the ouput can be used with <strong>ClusterGvis</strong>.</p>
<p>Here we run an example workflow with monocle3:</p>
<div class="sourceCode" id="cb73"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb73-1"><a href="parse-preudotime-data.html#cb73-1" tabindex="-1"></a><span class="co"># devtools::install_github(&#39;cole-trapnell-lab/monocle3&#39;)</span></span>
<span id="cb73-2"><a href="parse-preudotime-data.html#cb73-2" tabindex="-1"></a><span class="fu">library</span>(monocle3)</span>
<span id="cb73-3"><a href="parse-preudotime-data.html#cb73-3" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb73-4"><a href="parse-preudotime-data.html#cb73-4" tabindex="-1"></a></span>
<span id="cb73-5"><a href="parse-preudotime-data.html#cb73-5" tabindex="-1"></a>cell_metadata <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">system.file</span>(<span class="st">&#39;extdata&#39;</span>,</span>
<span id="cb73-6"><a href="parse-preudotime-data.html#cb73-6" tabindex="-1"></a>                                     <span class="st">&#39;worm_embryo/worm_embryo_coldata.rds&#39;</span>,</span>
<span id="cb73-7"><a href="parse-preudotime-data.html#cb73-7" tabindex="-1"></a>                                     <span class="at">package=</span><span class="st">&#39;monocle3&#39;</span>))</span>
<span id="cb73-8"><a href="parse-preudotime-data.html#cb73-8" tabindex="-1"></a>gene_metadata <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">system.file</span>(<span class="st">&#39;extdata&#39;</span>,</span>
<span id="cb73-9"><a href="parse-preudotime-data.html#cb73-9" tabindex="-1"></a>                                     <span class="st">&#39;worm_embryo/worm_embryo_rowdata.rds&#39;</span>,</span>
<span id="cb73-10"><a href="parse-preudotime-data.html#cb73-10" tabindex="-1"></a>                                     <span class="at">package=</span><span class="st">&#39;monocle3&#39;</span>))</span>
<span id="cb73-11"><a href="parse-preudotime-data.html#cb73-11" tabindex="-1"></a>expression_matrix <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">system.file</span>(<span class="st">&#39;extdata&#39;</span>,</span>
<span id="cb73-12"><a href="parse-preudotime-data.html#cb73-12" tabindex="-1"></a>                                         <span class="st">&#39;worm_embryo/worm_embryo_expression_matrix.rds&#39;</span>,</span>
<span id="cb73-13"><a href="parse-preudotime-data.html#cb73-13" tabindex="-1"></a>                                         <span class="at">package=</span><span class="st">&#39;monocle3&#39;</span>))</span>
<span id="cb73-14"><a href="parse-preudotime-data.html#cb73-14" tabindex="-1"></a></span>
<span id="cb73-15"><a href="parse-preudotime-data.html#cb73-15" tabindex="-1"></a>cds <span class="ot">&lt;-</span> <span class="fu">new_cell_data_set</span>(<span class="at">expression_data=</span>expression_matrix,</span>
<span id="cb73-16"><a href="parse-preudotime-data.html#cb73-16" tabindex="-1"></a>                         <span class="at">cell_metadata=</span>cell_metadata,</span>
<span id="cb73-17"><a href="parse-preudotime-data.html#cb73-17" tabindex="-1"></a>                         <span class="at">gene_metadata=</span>gene_metadata)</span>
<span id="cb73-18"><a href="parse-preudotime-data.html#cb73-18" tabindex="-1"></a></span>
<span id="cb73-19"><a href="parse-preudotime-data.html#cb73-19" tabindex="-1"></a>cds <span class="ot">&lt;-</span> <span class="fu">preprocess_cds</span>(cds)</span>
<span id="cb73-20"><a href="parse-preudotime-data.html#cb73-20" tabindex="-1"></a>cds <span class="ot">&lt;-</span> <span class="fu">align_cds</span>(cds, <span class="at">alignment_group =</span></span>
<span id="cb73-21"><a href="parse-preudotime-data.html#cb73-21" tabindex="-1"></a>                   <span class="st">&quot;batch&quot;</span>, <span class="at">residual_model_formula_str =</span> <span class="st">&quot;~ bg.300.loading +</span></span>
<span id="cb73-22"><a href="parse-preudotime-data.html#cb73-22" tabindex="-1"></a><span class="st">                      bg.400.loading + bg.500.1.loading + bg.500.2.loading +</span></span>
<span id="cb73-23"><a href="parse-preudotime-data.html#cb73-23" tabindex="-1"></a><span class="st">                      bg.r17.loading + bg.b01.loading + bg.b02.loading&quot;</span>)</span>
<span id="cb73-24"><a href="parse-preudotime-data.html#cb73-24" tabindex="-1"></a>cds <span class="ot">&lt;-</span> <span class="fu">reduce_dimension</span>(cds)</span>
<span id="cb73-25"><a href="parse-preudotime-data.html#cb73-25" tabindex="-1"></a>cds <span class="ot">&lt;-</span> <span class="fu">cluster_cells</span>(cds)</span>
<span id="cb73-26"><a href="parse-preudotime-data.html#cb73-26" tabindex="-1"></a>cds <span class="ot">&lt;-</span> <span class="fu">learn_graph</span>(cds)</span>
<span id="cb73-27"><a href="parse-preudotime-data.html#cb73-27" tabindex="-1"></a>cds <span class="ot">&lt;-</span> <span class="fu">order_cells</span>(cds, <span class="at">root_pr_nodes=</span><span class="st">&#39;Y_21&#39;</span>)</span>
<span id="cb73-28"><a href="parse-preudotime-data.html#cb73-28" tabindex="-1"></a></span>
<span id="cb73-29"><a href="parse-preudotime-data.html#cb73-29" tabindex="-1"></a>modulated_genes <span class="ot">&lt;-</span> <span class="fu">graph_test</span>(cds, <span class="at">neighbor_graph =</span> <span class="st">&quot;principal_graph&quot;</span>, <span class="at">cores =</span> <span class="dv">4</span>)</span>
<span id="cb73-30"><a href="parse-preudotime-data.html#cb73-30" tabindex="-1"></a>genes <span class="ot">&lt;-</span> <span class="fu">row.names</span>(<span class="fu">subset</span>(modulated_genes, q_value <span class="sc">==</span> <span class="dv">0</span> <span class="sc">&amp;</span> morans_I <span class="sc">&gt;</span> <span class="fl">0.25</span>))</span></code></pre></div>
<p>Extact expression matrix:</p>
<div class="sourceCode" id="cb74"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb74-1"><a href="parse-preudotime-data.html#cb74-1" tabindex="-1"></a>mat <span class="ot">&lt;-</span> <span class="fu">pre_pseudotime_matrix</span>(<span class="at">cds_obj =</span> cds,</span>
<span id="cb74-2"><a href="parse-preudotime-data.html#cb74-2" tabindex="-1"></a>                             <span class="at">gene_list =</span> genes)</span>
<span id="cb74-3"><a href="parse-preudotime-data.html#cb74-3" tabindex="-1"></a></span>
<span id="cb74-4"><a href="parse-preudotime-data.html#cb74-4" tabindex="-1"></a><span class="co"># check</span></span>
<span id="cb74-5"><a href="parse-preudotime-data.html#cb74-5" tabindex="-1"></a><span class="fu">head</span>(mat[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>,<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>])</span>
<span id="cb74-6"><a href="parse-preudotime-data.html#cb74-6" tabindex="-1"></a><span class="co">#                         1          2          3          4          5</span></span>
<span id="cb74-7"><a href="parse-preudotime-data.html#cb74-7" tabindex="-1"></a><span class="co"># WBGene00013882 -0.7265393 -0.7263892 -0.7262392 -0.7260891 -0.7259390</span></span>
<span id="cb74-8"><a href="parse-preudotime-data.html#cb74-8" tabindex="-1"></a><span class="co"># WBGene00002085  0.8059370  0.8076543  0.8093716  0.8110889  0.8128062</span></span>
<span id="cb74-9"><a href="parse-preudotime-data.html#cb74-9" tabindex="-1"></a><span class="co"># WBGene00016114  2.8256183  2.8198074  2.8139964  2.8081855  2.8023746</span></span>
<span id="cb74-10"><a href="parse-preudotime-data.html#cb74-10" tabindex="-1"></a><span class="co"># WBGene00012753  2.7917688  2.7861478  2.7805268  2.7749058  2.7692848</span></span>
<span id="cb74-11"><a href="parse-preudotime-data.html#cb74-11" tabindex="-1"></a><span class="co"># WBGene00015354 -1.1233255 -1.1229564 -1.1225872 -1.1222180 -1.1218489</span></span></code></pre></div>
<p>Run <strong>ClusterGvis</strong> workflow:</p>
<div class="sourceCode" id="cb75"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb75-1"><a href="parse-preudotime-data.html#cb75-1" tabindex="-1"></a><span class="co"># kmeans</span></span>
<span id="cb75-2"><a href="parse-preudotime-data.html#cb75-2" tabindex="-1"></a>ck <span class="ot">&lt;-</span> <span class="fu">clusterData</span>(<span class="at">exp =</span> mat,</span>
<span id="cb75-3"><a href="parse-preudotime-data.html#cb75-3" tabindex="-1"></a>                  <span class="at">cluster.method =</span> <span class="st">&quot;kmeans&quot;</span>,</span>
<span id="cb75-4"><a href="parse-preudotime-data.html#cb75-4" tabindex="-1"></a>                  <span class="at">cluster.num =</span> <span class="dv">5</span>)</span>
<span id="cb75-5"><a href="parse-preudotime-data.html#cb75-5" tabindex="-1"></a></span>
<span id="cb75-6"><a href="parse-preudotime-data.html#cb75-6" tabindex="-1"></a><span class="co"># add line annotation</span></span>
<span id="cb75-7"><a href="parse-preudotime-data.html#cb75-7" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">&#39;monocle3.pdf&#39;</span>,<span class="at">height =</span> <span class="dv">10</span>,<span class="at">width =</span> <span class="dv">8</span>,<span class="at">onefile =</span> F)</span>
<span id="cb75-8"><a href="parse-preudotime-data.html#cb75-8" tabindex="-1"></a><span class="fu">visCluster</span>(<span class="at">object =</span> ck,</span>
<span id="cb75-9"><a href="parse-preudotime-data.html#cb75-9" tabindex="-1"></a>           <span class="at">plot.type =</span> <span class="st">&quot;both&quot;</span>,</span>
<span id="cb75-10"><a href="parse-preudotime-data.html#cb75-10" tabindex="-1"></a>           <span class="at">add.sampleanno =</span> F,</span>
<span id="cb75-11"><a href="parse-preudotime-data.html#cb75-11" tabindex="-1"></a>           <span class="at">markGenes =</span> <span class="fu">sample</span>(<span class="fu">rownames</span>(mat),<span class="dv">30</span>,<span class="at">replace =</span> F))</span>
<span id="cb75-12"><a href="parse-preudotime-data.html#cb75-12" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
<p align="center">
<img src="local_images/45.png" width="80%">
</p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="parse-single-cell-data.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="add-custom-annotation.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/05-preudotime_parse.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
