<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 4 Parse WGCNA output | ClusterGVis Reference Book</title>
  <meta name="description" content="This is a detailed documentation of ClusterGVis package." />
  <meta name="generator" content="bookdown 0.41 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 4 Parse WGCNA output | ClusterGVis Reference Book" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This is a detailed documentation of ClusterGVis package." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 4 Parse WGCNA output | ClusterGVis Reference Book" />
  
  <meta name="twitter:description" content="This is a detailed documentation of ClusterGVis package." />
  

<meta name="author" content="Jun Zhang" />


<meta name="date" content="2025-05-15" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="enrichment-anaysis.html"/>
<link rel="next" href="parse-single-cell-data.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">ClusterGVis manual documentation</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> About</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#citation"><i class="fa fa-check"></i><b>1.1</b> Citation</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#installation"><i class="fa fa-check"></i><b>1.2</b> Installation</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#bugs"><i class="fa fa-check"></i><b>1.3</b> Bugs</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="basic-usage.html"><a href="basic-usage.html"><i class="fa fa-check"></i><b>2</b> Basic usage</a>
<ul>
<li class="chapter" data-level="2.1" data-path="basic-usage.html"><a href="basic-usage.html#input-data"><i class="fa fa-check"></i><b>2.1</b> Input data</a></li>
<li class="chapter" data-level="2.2" data-path="basic-usage.html"><a href="basic-usage.html#clustering-for-expression-matrix"><i class="fa fa-check"></i><b>2.2</b> Clustering for expression matrix</a></li>
<li class="chapter" data-level="2.3" data-path="basic-usage.html"><a href="basic-usage.html#line-plot"><i class="fa fa-check"></i><b>2.3</b> Line plot</a></li>
<li class="chapter" data-level="2.4" data-path="basic-usage.html"><a href="basic-usage.html#heatmap-plot"><i class="fa fa-check"></i><b>2.4</b> Heatmap plot</a></li>
<li class="chapter" data-level="2.5" data-path="basic-usage.html"><a href="basic-usage.html#sample-group-and-order"><i class="fa fa-check"></i><b>2.5</b> Sample group and order</a></li>
<li class="chapter" data-level="2.6" data-path="basic-usage.html"><a href="basic-usage.html#multiple-sample-annotation"><i class="fa fa-check"></i><b>2.6</b> Multiple sample annotation</a></li>
<li class="chapter" data-level="2.7" data-path="basic-usage.html"><a href="basic-usage.html#add-row-annotation"><i class="fa fa-check"></i><b>2.7</b> Add row annotation</a></li>
<li class="chapter" data-level="2.8" data-path="basic-usage.html"><a href="basic-usage.html#add-line-trend-anntation"><i class="fa fa-check"></i><b>2.8</b> Add line trend anntation</a></li>
<li class="chapter" data-level="2.9" data-path="basic-usage.html"><a href="basic-usage.html#multiple-group-line-plot"><i class="fa fa-check"></i><b>2.9</b> Multiple group line plot</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="enrichment-anaysis.html"><a href="enrichment-anaysis.html"><i class="fa fa-check"></i><b>3</b> Enrichment anaysis</a>
<ul>
<li class="chapter" data-level="3.1" data-path="enrichment-anaysis.html"><a href="enrichment-anaysis.html#enrichment-for-non-model-organisms"><i class="fa fa-check"></i><b>3.1</b> Enrichment for non-model organisms</a></li>
<li class="chapter" data-level="3.2" data-path="enrichment-anaysis.html"><a href="enrichment-anaysis.html#own-enrichment-data"><i class="fa fa-check"></i><b>3.2</b> Own enrichment data</a></li>
<li class="chapter" data-level="3.3" data-path="enrichment-anaysis.html"><a href="enrichment-anaysis.html#batch-enrichment-analysis"><i class="fa fa-check"></i><b>3.3</b> Batch enrichment analysis</a></li>
<li class="chapter" data-level="3.4" data-path="enrichment-anaysis.html"><a href="enrichment-anaysis.html#add-annotation-for-specific-groups"><i class="fa fa-check"></i><b>3.4</b> Add annotation for specific groups</a></li>
<li class="chapter" data-level="3.5" data-path="enrichment-anaysis.html"><a href="enrichment-anaysis.html#add-barplot-for-enrichment-annotation"><i class="fa fa-check"></i><b>3.5</b> Add barplot for enrichment annotation</a></li>
<li class="chapter" data-level="3.6" data-path="enrichment-anaysis.html"><a href="enrichment-anaysis.html#add-kegg-annotation"><i class="fa fa-check"></i><b>3.6</b> Add KEGG annotation</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="parse-wgcna-output.html"><a href="parse-wgcna-output.html"><i class="fa fa-check"></i><b>4</b> Parse WGCNA output</a></li>
<li class="chapter" data-level="5" data-path="parse-single-cell-data.html"><a href="parse-single-cell-data.html"><i class="fa fa-check"></i><b>5</b> Parse single cell data</a></li>
<li class="chapter" data-level="6" data-path="parse-preudotime-data.html"><a href="parse-preudotime-data.html"><i class="fa fa-check"></i><b>6</b> Parse preudotime data</a>
<ul>
<li class="chapter" data-level="6.1" data-path="parse-preudotime-data.html"><a href="parse-preudotime-data.html#monocle2-implemention"><i class="fa fa-check"></i><b>6.1</b> Monocle2 implemention</a>
<ul>
<li class="chapter" data-level="6.1.1" data-path="parse-preudotime-data.html"><a href="parse-preudotime-data.html#plot_pseudotime_heatmap2"><i class="fa fa-check"></i><b>6.1.1</b> plot_pseudotime_heatmap2</a></li>
<li class="chapter" data-level="6.1.2" data-path="parse-preudotime-data.html"><a href="parse-preudotime-data.html#plot_genes_branched_heatmap2"><i class="fa fa-check"></i><b>6.1.2</b> plot_genes_branched_heatmap2</a></li>
<li class="chapter" data-level="6.1.3" data-path="parse-preudotime-data.html"><a href="parse-preudotime-data.html#plot_multiple_branches_heatmap2"><i class="fa fa-check"></i><b>6.1.3</b> plot_multiple_branches_heatmap2</a></li>
</ul></li>
<li class="chapter" data-level="6.2" data-path="parse-preudotime-data.html"><a href="parse-preudotime-data.html#monocle3-implemention"><i class="fa fa-check"></i><b>6.2</b> Monocle3 implemention</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="add-custom-annotation.html"><a href="add-custom-annotation.html"><i class="fa fa-check"></i><b>7</b> Add custom annotation</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">ClusterGVis Reference Book</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="parse-wgcna-output" class="section level1 hasAnchor" number="4">
<h1><span class="header-section-number">Chapter 4</span> Parse WGCNA output<a href="parse-wgcna-output.html#parse-wgcna-output" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p><strong>WGCNA</strong> stands for Weighted Gene Co-expression Network Analysis. It’s a widely used bioinformatics method for analyzing high-throughput gene expression data. It will also cluster the expression data
into different clusters. <strong>ClusterGvis</strong> also can parse the <strong>WGCNA</strong> output results and analysis
biological significance for each cluster.</p>
<p>For detailed WGCNA tutorial, you can visit <a href="https://edo98811.github.io/WGCNA_official_documentation/">https://edo98811.github.io/WGCNA_official_documentation/</a>. <strong>ClusterGvis</strong> accepts an object from <strong>one-step network construction and module detection</strong></p>
<p>Related example code(<strong>not run</strong>):</p>
<div class="sourceCode" id="cb46"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb46-1"><a href="parse-wgcna-output.html#cb46-1" tabindex="-1"></a>net <span class="ot">=</span> <span class="fu">blockwiseModules</span>(datExpr, <span class="at">power =</span> <span class="dv">6</span>,</span>
<span id="cb46-2"><a href="parse-wgcna-output.html#cb46-2" tabindex="-1"></a>                       <span class="at">TOMType =</span> <span class="st">&quot;unsigned&quot;</span>, <span class="at">minModuleSize =</span> <span class="dv">30</span>,</span>
<span id="cb46-3"><a href="parse-wgcna-output.html#cb46-3" tabindex="-1"></a>                       <span class="at">reassignThreshold =</span> <span class="dv">0</span>, <span class="at">mergeCutHeight =</span> <span class="fl">0.25</span>,</span>
<span id="cb46-4"><a href="parse-wgcna-output.html#cb46-4" tabindex="-1"></a>                       <span class="at">numericLabels =</span> <span class="cn">TRUE</span>, <span class="at">pamRespectsDendro =</span> <span class="cn">FALSE</span>,</span>
<span id="cb46-5"><a href="parse-wgcna-output.html#cb46-5" tabindex="-1"></a>                       <span class="at">saveTOMs =</span> <span class="cn">TRUE</span>,</span>
<span id="cb46-6"><a href="parse-wgcna-output.html#cb46-6" tabindex="-1"></a>                       <span class="at">saveTOMFileBase =</span> <span class="st">&quot;femaleMouseTOM&quot;</span>,</span>
<span id="cb46-7"><a href="parse-wgcna-output.html#cb46-7" tabindex="-1"></a>                       <span class="at">verbose =</span> <span class="dv">3</span>)</span></code></pre></div>
<p>Pass to <strong>clusterData</strong> to obtain plotting data. Note that the <strong>exp</strong> provided here should be the expression matrix after WGCNA filtering. It’s a transposed matrix, corresponding to the <strong>datExpr</strong> in the WGCNA example code:</p>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="parse-wgcna-output.html#cb47-1" tabindex="-1"></a>datExpr0 <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">t</span>(femData[, <span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>)]));</span>
<span id="cb47-2"><a href="parse-wgcna-output.html#cb47-2" tabindex="-1"></a><span class="fu">names</span>(datExpr0) <span class="ot">=</span> femData<span class="sc">$</span>substanceBXH;</span>
<span id="cb47-3"><a href="parse-wgcna-output.html#cb47-3" tabindex="-1"></a><span class="fu">rownames</span>(datExpr0) <span class="ot">=</span> <span class="fu">names</span>(femData)[<span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>)];</span>
<span id="cb47-4"><a href="parse-wgcna-output.html#cb47-4" tabindex="-1"></a></span>
<span id="cb47-5"><a href="parse-wgcna-output.html#cb47-5" tabindex="-1"></a>gsg <span class="ot">=</span> <span class="fu">goodSamplesGenes</span>(datExpr0, <span class="at">verbose =</span> <span class="dv">3</span>);</span>
<span id="cb47-6"><a href="parse-wgcna-output.html#cb47-6" tabindex="-1"></a>gsg<span class="sc">$</span>allOK</span>
<span id="cb47-7"><a href="parse-wgcna-output.html#cb47-7" tabindex="-1"></a></span>
<span id="cb47-8"><a href="parse-wgcna-output.html#cb47-8" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span>gsg<span class="sc">$</span>allOK)</span>
<span id="cb47-9"><a href="parse-wgcna-output.html#cb47-9" tabindex="-1"></a>{</span>
<span id="cb47-10"><a href="parse-wgcna-output.html#cb47-10" tabindex="-1"></a>  <span class="co"># Optionally, print the gene and sample names that were removed:</span></span>
<span id="cb47-11"><a href="parse-wgcna-output.html#cb47-11" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">sum</span>(<span class="sc">!</span>gsg<span class="sc">$</span>goodGenes)<span class="sc">&gt;</span><span class="dv">0</span>)</span>
<span id="cb47-12"><a href="parse-wgcna-output.html#cb47-12" tabindex="-1"></a>     <span class="fu">printFlush</span>(<span class="fu">paste</span>(<span class="st">&quot;Removing genes:&quot;</span>, <span class="fu">paste</span>(<span class="fu">names</span>(datExpr0)[<span class="sc">!</span>gsg<span class="sc">$</span>goodGenes], <span class="at">collapse =</span> <span class="st">&quot;, &quot;</span>)));</span>
<span id="cb47-13"><a href="parse-wgcna-output.html#cb47-13" tabindex="-1"></a>  <span class="cf">if</span> (<span class="fu">sum</span>(<span class="sc">!</span>gsg<span class="sc">$</span>goodSamples)<span class="sc">&gt;</span><span class="dv">0</span>)</span>
<span id="cb47-14"><a href="parse-wgcna-output.html#cb47-14" tabindex="-1"></a>     <span class="fu">printFlush</span>(<span class="fu">paste</span>(<span class="st">&quot;Removing samples:&quot;</span>, <span class="fu">paste</span>(<span class="fu">rownames</span>(datExpr0)[<span class="sc">!</span>gsg<span class="sc">$</span>goodSamples], <span class="at">collapse =</span> <span class="st">&quot;, &quot;</span>)));</span>
<span id="cb47-15"><a href="parse-wgcna-output.html#cb47-15" tabindex="-1"></a>  <span class="co"># Remove the offending genes and samples from the data:</span></span>
<span id="cb47-16"><a href="parse-wgcna-output.html#cb47-16" tabindex="-1"></a>  datExpr0 <span class="ot">=</span> datExpr0[gsg<span class="sc">$</span>goodSamples, gsg<span class="sc">$</span>goodGenes]</span>
<span id="cb47-17"><a href="parse-wgcna-output.html#cb47-17" tabindex="-1"></a>}</span>
<span id="cb47-18"><a href="parse-wgcna-output.html#cb47-18" tabindex="-1"></a></span>
<span id="cb47-19"><a href="parse-wgcna-output.html#cb47-19" tabindex="-1"></a>sampleTree <span class="ot">=</span> <span class="fu">hclust</span>(<span class="fu">dist</span>(datExpr0), <span class="at">method =</span> <span class="st">&quot;average&quot;</span>);</span>
<span id="cb47-20"><a href="parse-wgcna-output.html#cb47-20" tabindex="-1"></a><span class="co"># Plot the sample tree: Open a graphic output window of size 12 by 9 inches</span></span>
<span id="cb47-21"><a href="parse-wgcna-output.html#cb47-21" tabindex="-1"></a><span class="co"># The user should change the dimensions if the window is too large or too small.</span></span>
<span id="cb47-22"><a href="parse-wgcna-output.html#cb47-22" tabindex="-1"></a><span class="fu">sizeGrWindow</span>(<span class="dv">12</span>,<span class="dv">9</span>)</span>
<span id="cb47-23"><a href="parse-wgcna-output.html#cb47-23" tabindex="-1"></a><span class="co">#pdf(file = &quot;Plots/sampleClustering.pdf&quot;, width = 12, height = 9);</span></span>
<span id="cb47-24"><a href="parse-wgcna-output.html#cb47-24" tabindex="-1"></a><span class="fu">par</span>(<span class="at">cex =</span> <span class="fl">0.6</span>);</span>
<span id="cb47-25"><a href="parse-wgcna-output.html#cb47-25" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mar =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">4</span>,<span class="dv">2</span>,<span class="dv">0</span>))</span>
<span id="cb47-26"><a href="parse-wgcna-output.html#cb47-26" tabindex="-1"></a><span class="fu">plot</span>(sampleTree, <span class="at">main =</span> <span class="st">&quot;Sample clustering to detect outliers&quot;</span>, <span class="at">sub=</span><span class="st">&quot;&quot;</span>, <span class="at">xlab=</span><span class="st">&quot;&quot;</span>, <span class="at">cex.lab =</span> <span class="fl">1.5</span>,</span>
<span id="cb47-27"><a href="parse-wgcna-output.html#cb47-27" tabindex="-1"></a>     <span class="at">cex.axis =</span> <span class="fl">1.5</span>, <span class="at">cex.main =</span> <span class="dv">2</span>)</span>
<span id="cb47-28"><a href="parse-wgcna-output.html#cb47-28" tabindex="-1"></a></span>
<span id="cb47-29"><a href="parse-wgcna-output.html#cb47-29" tabindex="-1"></a><span class="co"># Plot a line to show the cut</span></span>
<span id="cb47-30"><a href="parse-wgcna-output.html#cb47-30" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h =</span> <span class="dv">15</span>, <span class="at">col =</span> <span class="st">&quot;red&quot;</span>);</span>
<span id="cb47-31"><a href="parse-wgcna-output.html#cb47-31" tabindex="-1"></a><span class="co"># Determine cluster under the line</span></span>
<span id="cb47-32"><a href="parse-wgcna-output.html#cb47-32" tabindex="-1"></a>clust <span class="ot">=</span> <span class="fu">cutreeStatic</span>(sampleTree, <span class="at">cutHeight =</span> <span class="dv">15</span>, <span class="at">minSize =</span> <span class="dv">10</span>)</span>
<span id="cb47-33"><a href="parse-wgcna-output.html#cb47-33" tabindex="-1"></a><span class="fu">table</span>(clust)</span>
<span id="cb47-34"><a href="parse-wgcna-output.html#cb47-34" tabindex="-1"></a><span class="co"># clust 1 contains the samples we want to keep.</span></span>
<span id="cb47-35"><a href="parse-wgcna-output.html#cb47-35" tabindex="-1"></a>keepSamples <span class="ot">=</span> (clust<span class="sc">==</span><span class="dv">1</span>)</span>
<span id="cb47-36"><a href="parse-wgcna-output.html#cb47-36" tabindex="-1"></a>datExpr <span class="ot">=</span> datExpr0[keepSamples, ]</span></code></pre></div>
<p>We load a test WGCNA object which ran with WGCNA using <strong>exps</strong> data:</p>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="parse-wgcna-output.html#cb48-1" tabindex="-1"></a><span class="fu">data</span>(<span class="st">&quot;net&quot;</span>)</span>
<span id="cb48-2"><a href="parse-wgcna-output.html#cb48-2" tabindex="-1"></a><span class="fu">str</span>(net)</span>
<span id="cb48-3"><a href="parse-wgcna-output.html#cb48-3" tabindex="-1"></a></span>
<span id="cb48-4"><a href="parse-wgcna-output.html#cb48-4" tabindex="-1"></a><span class="co"># List of 10</span></span>
<span id="cb48-5"><a href="parse-wgcna-output.html#cb48-5" tabindex="-1"></a><span class="co"># $ colors        : Named num [1:3767] 1 1 1 1 1 1 1 1 1 1 ...</span></span>
<span id="cb48-6"><a href="parse-wgcna-output.html#cb48-6" tabindex="-1"></a><span class="co"># ..- attr(*, &quot;names&quot;)= chr [1:3767] &quot;Oog4&quot; &quot;Psmd9&quot; &quot;Sephs2&quot; &quot;Nhlrc2&quot; ...</span></span>
<span id="cb48-7"><a href="parse-wgcna-output.html#cb48-7" tabindex="-1"></a><span class="co"># $ unmergedColors: Named num [1:3767] 1 1 1 1 1 1 1 1 1 1 ...</span></span>
<span id="cb48-8"><a href="parse-wgcna-output.html#cb48-8" tabindex="-1"></a><span class="co"># ..- attr(*, &quot;names&quot;)= chr [1:3767] &quot;Oog4&quot; &quot;Psmd9&quot; &quot;Sephs2&quot; &quot;Nhlrc2&quot; ...</span></span>
<span id="cb48-9"><a href="parse-wgcna-output.html#cb48-9" tabindex="-1"></a><span class="co"># $ MEs           :&#39;data.frame&#39;: 6 obs. of  5 variables:</span></span>
<span id="cb48-10"><a href="parse-wgcna-output.html#cb48-10" tabindex="-1"></a><span class="co">#   ..$ ME1: num [1:6] -0.225 -0.251 -0.236 -0.196 0.018 ...</span></span>
<span id="cb48-11"><a href="parse-wgcna-output.html#cb48-11" tabindex="-1"></a><span class="co"># ..$ ME2: num [1:6] 0.564 0.423 0.103 -0.113 -0.456 ...</span></span>
<span id="cb48-12"><a href="parse-wgcna-output.html#cb48-12" tabindex="-1"></a><span class="co"># ..$ ME3: num [1:6] 0.228 0.448 0.107 -0.281 -0.766 ...</span></span>
<span id="cb48-13"><a href="parse-wgcna-output.html#cb48-13" tabindex="-1"></a><span class="co"># ..$ ME4: num [1:6] 0.62653 -0.00404 -0.26899 -0.44759 -0.35935 ...</span></span>
<span id="cb48-14"><a href="parse-wgcna-output.html#cb48-14" tabindex="-1"></a><span class="co"># ..$ ME0: num [1:6] -0.497 0.461 -0.162 0.587 -0.411 ...</span></span>
<span id="cb48-15"><a href="parse-wgcna-output.html#cb48-15" tabindex="-1"></a><span class="co"># $ goodSamples   : logi [1:6] TRUE TRUE TRUE TRUE TRUE TRUE</span></span>
<span id="cb48-16"><a href="parse-wgcna-output.html#cb48-16" tabindex="-1"></a><span class="co"># $ goodGenes     : logi [1:3767] TRUE TRUE TRUE TRUE TRUE TRUE ...</span></span>
<span id="cb48-17"><a href="parse-wgcna-output.html#cb48-17" tabindex="-1"></a><span class="co"># $ dendrograms   :List of 1</span></span>
<span id="cb48-18"><a href="parse-wgcna-output.html#cb48-18" tabindex="-1"></a><span class="co"># ..$ :List of 7</span></span>
<span id="cb48-19"><a href="parse-wgcna-output.html#cb48-19" tabindex="-1"></a><span class="co"># .. ..$ merge      : int [1:3766, 1:2] -1089 -1087 -1091 -377 -1092 -1104 -1054 -1095 -1088 -1064 ...</span></span>
<span id="cb48-20"><a href="parse-wgcna-output.html#cb48-20" tabindex="-1"></a><span class="co"># .. ..$ height     : num [1:3766] 0.19 0.19 0.19 0.19 0.19 ...</span></span>
<span id="cb48-21"><a href="parse-wgcna-output.html#cb48-21" tabindex="-1"></a><span class="co"># .. ..$ order      : int [1:3767] 2953 3637 3065 3270 3529 3660 3018 3600 3558 3665 ...</span></span>
<span id="cb48-22"><a href="parse-wgcna-output.html#cb48-22" tabindex="-1"></a><span class="co"># .. ..$ labels     : NULL</span></span>
<span id="cb48-23"><a href="parse-wgcna-output.html#cb48-23" tabindex="-1"></a><span class="co"># .. ..$ method     : chr &quot;average&quot;</span></span>
<span id="cb48-24"><a href="parse-wgcna-output.html#cb48-24" tabindex="-1"></a><span class="co"># .. ..$ call       : language fastcluster::hclust(d = as.dist(dissTom), method = &quot;average&quot;)</span></span>
<span id="cb48-25"><a href="parse-wgcna-output.html#cb48-25" tabindex="-1"></a><span class="co"># .. ..$ dist.method: NULL</span></span>
<span id="cb48-26"><a href="parse-wgcna-output.html#cb48-26" tabindex="-1"></a><span class="co"># .. ..- attr(*, &quot;class&quot;)= chr &quot;hclust&quot;</span></span>
<span id="cb48-27"><a href="parse-wgcna-output.html#cb48-27" tabindex="-1"></a><span class="co"># $ TOMFiles      : chr &quot;femaleMouseTOM-block.1.RData&quot;</span></span>
<span id="cb48-28"><a href="parse-wgcna-output.html#cb48-28" tabindex="-1"></a><span class="co"># $ blockGenes    :List of 1</span></span>
<span id="cb48-29"><a href="parse-wgcna-output.html#cb48-29" tabindex="-1"></a><span class="co"># ..$ : int [1:3767] 1 2 3 4 5 6 7 8 9 10 ...</span></span>
<span id="cb48-30"><a href="parse-wgcna-output.html#cb48-30" tabindex="-1"></a><span class="co"># $ blocks        : num [1:3767] 1 1 1 1 1 1 1 1 1 1 ...</span></span>
<span id="cb48-31"><a href="parse-wgcna-output.html#cb48-31" tabindex="-1"></a><span class="co"># $ MEsOK         : logi TRUE</span></span></code></pre></div>
<p>Parse the object:</p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="parse-wgcna-output.html#cb49-1" tabindex="-1"></a>expt <span class="ot">&lt;-</span> <span class="fu">t</span>(exps)</span>
<span id="cb49-2"><a href="parse-wgcna-output.html#cb49-2" tabindex="-1"></a></span>
<span id="cb49-3"><a href="parse-wgcna-output.html#cb49-3" tabindex="-1"></a><span class="co"># get data</span></span>
<span id="cb49-4"><a href="parse-wgcna-output.html#cb49-4" tabindex="-1"></a>cw <span class="ot">&lt;-</span> <span class="fu">clusterData</span>(<span class="at">obj =</span> expt,</span>
<span id="cb49-5"><a href="parse-wgcna-output.html#cb49-5" tabindex="-1"></a>                  <span class="at">cluster.method =</span> <span class="st">&quot;wgcna&quot;</span>,</span>
<span id="cb49-6"><a href="parse-wgcna-output.html#cb49-6" tabindex="-1"></a>                  <span class="at">object =</span> net)</span>
<span id="cb49-7"><a href="parse-wgcna-output.html#cb49-7" tabindex="-1"></a></span>
<span id="cb49-8"><a href="parse-wgcna-output.html#cb49-8" tabindex="-1"></a><span class="fu">str</span>(cw)</span>
<span id="cb49-9"><a href="parse-wgcna-output.html#cb49-9" tabindex="-1"></a><span class="co"># List of 3</span></span>
<span id="cb49-10"><a href="parse-wgcna-output.html#cb49-10" tabindex="-1"></a><span class="co"># $ wide.res:&#39;data.frame&#39;: 3767 obs. of  9 variables:</span></span>
<span id="cb49-11"><a href="parse-wgcna-output.html#cb49-11" tabindex="-1"></a><span class="co">#   ..$ zygote    : num [1:3767] 0.6747 0.3143 -0.0799 -0.1841 0.7904 ...</span></span>
<span id="cb49-12"><a href="parse-wgcna-output.html#cb49-12" tabindex="-1"></a><span class="co"># ..$ t2.cell   : num [1:3767] 0.511 1.083 1.14 0.497 -0.247 ...</span></span>
<span id="cb49-13"><a href="parse-wgcna-output.html#cb49-13" tabindex="-1"></a><span class="co"># ..$ t4.cell   : num [1:3767] 0.702 0.598 0.698 0.794 0.668 ...</span></span>
<span id="cb49-14"><a href="parse-wgcna-output.html#cb49-14" tabindex="-1"></a><span class="co"># ..$ t8.cell   : num [1:3767] 0.565 0.222 0.48 0.985 0.826 ...</span></span>
<span id="cb49-15"><a href="parse-wgcna-output.html#cb49-15" tabindex="-1"></a><span class="co"># ..$ tmorula   : num [1:3767] -0.743 -0.45 -0.636 -0.363 -0.273 ...</span></span>
<span id="cb49-16"><a href="parse-wgcna-output.html#cb49-16" tabindex="-1"></a><span class="co"># ..$ blastocyst: num [1:3767] -1.71 -1.77 -1.6 -1.73 -1.76 ...</span></span>
<span id="cb49-17"><a href="parse-wgcna-output.html#cb49-17" tabindex="-1"></a><span class="co"># ..$ gene      : chr [1:3767] &quot;Oog4&quot; &quot;Psmd9&quot; &quot;Sephs2&quot; &quot;Nhlrc2&quot; ...</span></span>
<span id="cb49-18"><a href="parse-wgcna-output.html#cb49-18" tabindex="-1"></a><span class="co"># ..$ cluster   : num [1:3767] 2 2 2 2 2 2 2 2 2 2 ...</span></span>
<span id="cb49-19"><a href="parse-wgcna-output.html#cb49-19" tabindex="-1"></a><span class="co"># ..$ modulecol : chr [1:3767] &quot;turquoise&quot; &quot;turquoise&quot; &quot;turquoise&quot; &quot;turquoise&quot; ...</span></span>
<span id="cb49-20"><a href="parse-wgcna-output.html#cb49-20" tabindex="-1"></a><span class="co"># $ long.res:&#39;data.frame&#39;: 22602 obs. of  6 variables:</span></span>
<span id="cb49-21"><a href="parse-wgcna-output.html#cb49-21" tabindex="-1"></a><span class="co">#   ..$ cluster     : num [1:22602] 2 2 2 2 2 2 2 2 2 2 ...</span></span>
<span id="cb49-22"><a href="parse-wgcna-output.html#cb49-22" tabindex="-1"></a><span class="co"># ..$ gene        : chr [1:22602] &quot;Oog4&quot; &quot;Psmd9&quot; &quot;Sephs2&quot; &quot;Nhlrc2&quot; ...</span></span>
<span id="cb49-23"><a href="parse-wgcna-output.html#cb49-23" tabindex="-1"></a><span class="co"># ..$ modulecol   : chr [1:22602] &quot;turquoise&quot; &quot;turquoise&quot; &quot;turquoise&quot; &quot;turquoise&quot; ...</span></span>
<span id="cb49-24"><a href="parse-wgcna-output.html#cb49-24" tabindex="-1"></a><span class="co"># ..$ cell_type   : Factor w/ 6 levels &quot;zygote&quot;,&quot;t2.cell&quot;,..: 1 1 1 1 1 1 1 1 1 1 ...</span></span>
<span id="cb49-25"><a href="parse-wgcna-output.html#cb49-25" tabindex="-1"></a><span class="co"># ..$ norm_value  : num [1:22602] 0.6747 0.3143 -0.0799 -0.1841 0.7904 ...</span></span>
<span id="cb49-26"><a href="parse-wgcna-output.html#cb49-26" tabindex="-1"></a><span class="co"># ..$ cluster_name: chr [1:22602] &quot;cluster 2 (2574 turquoise)&quot; &quot;cluster 2 (2574 turquoise)&quot; &quot;cluster 2 (2574 turquoise)&quot; &quot;cluster 2 (2574 turquoise)&quot; ...</span></span>
<span id="cb49-27"><a href="parse-wgcna-output.html#cb49-27" tabindex="-1"></a><span class="co"># $ type    : chr &quot;wgcna&quot;</span></span></code></pre></div>
<p>Now we can visualize the parsed data by using <strong>ClusterGvis</strong>:</p>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="parse-wgcna-output.html#cb50-1" tabindex="-1"></a><span class="co"># add gene name</span></span>
<span id="cb50-2"><a href="parse-wgcna-output.html#cb50-2" tabindex="-1"></a>markGenes <span class="ot">=</span> <span class="fu">rownames</span>(exps)[<span class="fu">sample</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(exps),<span class="dv">30</span>,<span class="at">replace =</span> F)]</span>
<span id="cb50-3"><a href="parse-wgcna-output.html#cb50-3" tabindex="-1"></a></span>
<span id="cb50-4"><a href="parse-wgcna-output.html#cb50-4" tabindex="-1"></a><span class="co"># plot</span></span>
<span id="cb50-5"><a href="parse-wgcna-output.html#cb50-5" tabindex="-1"></a><span class="fu">visCluster</span>(<span class="at">object =</span> cw,</span>
<span id="cb50-6"><a href="parse-wgcna-output.html#cb50-6" tabindex="-1"></a>           <span class="at">plot.type =</span> <span class="st">&quot;line&quot;</span>,</span>
<span id="cb50-7"><a href="parse-wgcna-output.html#cb50-7" tabindex="-1"></a>           <span class="at">ncol =</span> <span class="dv">3</span>)</span></code></pre></div>
<p align="center">
<img src="local_images/23.png" width="60%">
</p>
<p>Heatmap:</p>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="parse-wgcna-output.html#cb51-1" tabindex="-1"></a><span class="co"># heatmap</span></span>
<span id="cb51-2"><a href="parse-wgcna-output.html#cb51-2" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">&#39;netht.pdf&#39;</span>,<span class="at">height =</span> <span class="dv">10</span>,<span class="at">width =</span> <span class="dv">6</span>,<span class="at">onefile =</span> F)</span>
<span id="cb51-3"><a href="parse-wgcna-output.html#cb51-3" tabindex="-1"></a><span class="fu">visCluster</span>(<span class="at">object =</span> cw,</span>
<span id="cb51-4"><a href="parse-wgcna-output.html#cb51-4" tabindex="-1"></a>           <span class="at">plot.type =</span> <span class="st">&quot;heatmap&quot;</span>,</span>
<span id="cb51-5"><a href="parse-wgcna-output.html#cb51-5" tabindex="-1"></a>           <span class="at">markGenes =</span> markGenes)</span>
<span id="cb51-6"><a href="parse-wgcna-output.html#cb51-6" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
<p align="center">
<img src="local_images/24.png" width="60%">
</p>
<p>Add line annotation:</p>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="parse-wgcna-output.html#cb52-1" tabindex="-1"></a><span class="co"># heatmap + line</span></span>
<span id="cb52-2"><a href="parse-wgcna-output.html#cb52-2" tabindex="-1"></a><span class="fu">pdf</span>(<span class="st">&#39;nethtl.pdf&#39;</span>,<span class="at">height =</span> <span class="dv">10</span>,<span class="at">width =</span> <span class="dv">8</span>,<span class="at">onefile =</span> F)</span>
<span id="cb52-3"><a href="parse-wgcna-output.html#cb52-3" tabindex="-1"></a><span class="fu">visCluster</span>(<span class="at">object =</span> cw,</span>
<span id="cb52-4"><a href="parse-wgcna-output.html#cb52-4" tabindex="-1"></a>           <span class="at">plot.type =</span> <span class="st">&quot;both&quot;</span>,</span>
<span id="cb52-5"><a href="parse-wgcna-output.html#cb52-5" tabindex="-1"></a>           <span class="at">markGenes =</span> markGenes)</span>
<span id="cb52-6"><a href="parse-wgcna-output.html#cb52-6" tabindex="-1"></a><span class="fu">dev.off</span>()</span></code></pre></div>
<p align="center">
<img src="local_images/25.png" width="60%">
</p>

</div>
            </section>

          </div>
        </div>
      </div>
<a href="enrichment-anaysis.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="parse-single-cell-data.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/USERNAME/REPO/edit/BRANCH/03-WGCNA_parse.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
